easybuild/tools/options.py

```diff
@@ -405,6 +409,11 @@ def postprocess(self):
             if token is None:
                 self.log.error("Failed to obtain required GitHub token for user '%s'" % self.options.github_user)
 
+        # options w.r.t. updating Lmod cache only make sense when Lmod is selected modules tool
+        if self.options.update_lmod_caches:
+            if self.options.modules_tool != Lmod.__name__:
```

Comment:
should be a call to the modules tool class like `klass.can_create_cache` or something like that